### PR TITLE
fix: add status guards to Timings buttons and extend in-progress reallocation to subscriptions (#482)

### PR DESCRIPTION
**Pull request by shubham79a**
## Summary

- **Status guards**: Hide the "Timings" button on completed groups and completed/cancelled individual appointments, preventing slot reallocation on finished services

- **Subscription reallocation parity**: Extend in-progress reallocation protections (past slot preservation, required slot count adjustment, duplicate allocation guard) from classes to subscriptions across 6 locations

- **Tests**: Add 2 new test cases mirroring existing class tests for subscription in-progress reallocation 

#482

### CODE REVIEW

**Comment by shubham79a** [comment]
### Code review follow-ups (PR #479)
  - **Extract magic number**: Replace duplicated `24 * 60 * 60 * 1000` with shared `TWENTY_FOUR_HOURS_IN_MS` constant in `slotTimeUtils.ts`, imported by `SlotAllocationService.ts` and `UnifiedCalendar.tsx`    
  - **Remove `as any` cast**: Remove unnecessary `(slot as any).user` in `SlotAllocationService.ts` — Prisma's    
  inferred type from the `include` already provides the `user` array

File: app/dashboard/consultant/[consultantId]/(features)/shared/utils/allocationAlgorithms.ts [hunk 1]
```diff
@@ -97,7 +97,7 @@ export class AllocationAlgorithms {
       // For in-progress recurring events, subtract past confirmed slots
       const pastCount = options.pastConfirmedSlotCount || 0;
       const requiredSlots =
-        options.eventType === "class" && pastCount > 0
+        (options.eventType === "class" || options.eventType === "subscription") && pastCount > 0
           ? Math.max(0, rawRequired - pastCount)
           : rawRequired;
```
[the remark this answers is not in the document]
Comment (shubham79a) [reply app/dashboard/consultant/[consultantId]/(features)/shared/utils/allocationAlgorithms.ts:103]: https://github.com/Practitionist/familiarise_web/pull/483/changes/88dfd1169630952cb46b31607bdf093a46553114

File: app/dashboard/consultant/[consultantId]/(features)/appointments/components/EventTimingsCalendar.tsx [hunk 2]
```diff
@@ -125,16 +129,27 @@ export function EventTimingsCalendar({
...
-        return `Schedule ${meetingsPerWeek} meeting${meetingsPerWeek !== 1 ? "s" : ""} per week. Each session is ${durationText}.`;
+        const classBaseText = `Schedule ${meetingsPerWeek} meeting${meetingsPerWeek !== 1 ? "s" : ""} per week. Each session is ${durationText}.`;
+        if (completedSessions && completedSessions > 0 && groupTotalSessions) {
+          const remaining = groupTotalSessions - completedSessions;
+          return `${classBaseText} ${completedSessions} of ${groupTotalSessions} sessions completed — select times for the remaining ${remaining}.`;
+        }
+        return classBaseText;
       }
       default:
         return "Select time slots for your event.";
```
[the remark this answers is not in the document]
Comment (shubham79a) [reply app/dashboard/consultant/[consultantId]/(features)/appointments/components/EventTimingsCalendar.tsx:159]: https://github.com/Practitionist/familiarise_web/pull/483/changes/8a9710e1b9d4133a5c00c219c9456a0dc1907a45

File: __tests__/booking-algorithm/slotAllocationService.test.ts [hunk 3]
```diff
@@ -1476,8 +1477,7 @@ describe("Edge cases", () => {
         eventFactories[eventType](),
       );
 
-      const slots =
-        eventType === "subscription" || eventType === "class"
+      const slots = isRecurringEventType(eventType)
           ? ["2025-01-06T10:00:00Z", "2025-01-06T10:30:00Z"]
           : ["2025-01-06T10:00:00Z", "2025-01-06T10:30:00Z"];
```
[the remark this answers is not in the document]
Comment (shubham79a) [reply __tests__/booking-algorithm/slotAllocationService.test.ts]: https://github.com/Practitionist/familiarise_web/pull/483/changes/8a9710e1b9d4133a5c00c219c9456a0dc1907a45
